### PR TITLE
feat(profile): add profile page UI and base functionality

### PR DESCRIPTION
**Pull request by anette1983**
## Describe your changes

- Created Profile page UI

Знімок екрана 2023-11-29 163328
Знімок екрана 2023-11-29 163454
Знімок екрана 2023-11-29 163550

## Issue ticket code (and/or) and link

- [Link to JIRA ticket](https://testhomework4.atlassian.net/jira/software/projects/AG/boards/6?selectedIssue=AG-131)

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [ ] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [ ] Date and time formats are on the constants
- [ ] Functions are public only if it's used outside the class
- [ ] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.

### CODE REVIEW

File: src/sections/user/profile-cover.tsx [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+import Avatar from '@mui/material/Avatar';
+import ListItemText from '@mui/material/ListItemText';
+import { useTheme, alpha } from '@mui/material/styles';
+import { IUserProfileCover } from 'types/user';
+import { bgGradient } from 'theme/css';
+import AvatarShape from 'assets/illustrations/avatar-shape';
+import Iconify from 'components/iconify';
+import { links } from 'constants/links';
+
+// ----------------------------------------------------------------------
```
Comment (AKutusenko) [inline src/sections/user/profile-cover.tsx]: Useless comment string

File: src/sections/user/view/user-profile-view.tsx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    icon: <Iconify icon="solar:user-id-bold" width={24} />,
+  },
+  {
+    value: 'settings',
+    label: 'Settings',
+    icon: <Iconify icon="solar:settings-bold" width={24} />,
+  },
+];
+
+// ----------------------------------------------------------------------
```
Comment (AKutusenko) [inline src/sections/user/view/user-profile-view.tsx]: Same

File: src/sections/user/profile-cover.tsx (hunk 1 above)
Comment (anette1983) [reply src/sections/user/profile-cover.tsx]: fixed

File: src/sections/user/view/user-profile-view.tsx (hunk 2 above)
Comment (anette1983) [reply src/sections/user/view/user-profile-view.tsx]: fixed

File: src/types/langs.ts [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
-  };
-};
+// type SignUpPageType = {
+//   Main: {
+//     [key: string]: string;
+//   };
+//   TermsContent: {
+//     [key: string]: string;
+//   };
+// };
```
Comment (AKutusenko) [inline src/types/langs.ts]: delete commented code if it's not part of documentation

File: src/types/langs.ts [hunk 4]
```diff
@@ -1,12 +1,12 @@
...
 
-type SignUpProps = {
-  SignUpPage: SignUpPageType;
-};
+// type SignUpProps = {
+//   SignUpPage: SignUpPageType;
+// };
```
Comment (AKutusenko) [inline src/types/langs.ts]: same

File: src/types/langs.ts (hunk 3 above)
Comment (anette1983) [reply src/types/langs.ts]: fixed

File: src/types/langs.ts (hunk 4 above)
Comment (anette1983) [reply src/types/langs.ts]: fixed

**Review by AKutusenko — APPROVED** [review]

File: src/sections/user/profile-home.tsx [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+            <Iconify
+              icon={link.icon}
+              width={24}
+              sx={{
+                flexShrink: 0,
+                color: link.color,
+              }}
+            />
+            <Link color="inherit">
+              {link.value === 'facebook' && info.socialLinks.facebook}
```
Comment (lyudad) [inline src/sections/user/profile-home.tsx]: https://www.notion.so/Git-rules-849f8ca8b3db4ded926d762c6fff8798 1.1.5

**Review by lyudad — APPROVED** [review]

**Pull request merged**